pandas/core/internals/blocks.py

```diff
@@ -530,6 +530,10 @@ def f(m, v, i):
         return self.split_and_operate(None, f, False)
 
     def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
+        dtype = pandas_dtype(dtype)
```

Comment:
i don’t want to put this clause where the existing is?

Reply:
the issue here is that we do all of the work in _astype, so now this is obscuring things